### PR TITLE
[DOC] Updates for config exceptions

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

- Added a description of `zookeeper.connection.timeout.ms` for Kafka broker config
- Added a description of `ssl.endpoint.identification.algorithm` for:
    - KafkaConnect
    - KafkaConnectS2I
    - MirrorMaker consumer and producer
    - MirrorMaker2
    - Kafka bridge consumer and producer

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by PaulRMellor** [comment]
This PR fixes https://github.com/strimzi/strimzi-kafka-operator/issues/1954

File: documentation/modules/ref-kafka-connect-configuration.adoc [hunk 1]
```diff
@@ -42,8 +42,10 @@ Certain options have default values:
 
 These options are automatically configured in case they are not present in the `KafkaConnect.spec.config` or `KafkaConnectS2I.spec.config` properties.
 
-Use the three allowed `ssl` configuration options for client connection using a specific _cipher suite_ for a TLS version.
+There are exceptions to the restrictions imposed on the configuration.
```
Comment (laidan6000) [inline documentation/modules/ref-kafka-connect-configuration.adoc:45]: _Suggested change:_

There are exceptions to the restrictions imposed on the configuration > There are exceptions to the forbidden options

Clarify that the exceptions are related to the list of "forbidden" options above. I was not quite sure at first.

File: documentation/modules/ref-kafka-broker-configuration.adoc [hunk 2]
```diff
@@ -35,8 +35,10 @@ Specifically, you are prevented from modifying all configuration options with ke
 If the `config` property specifies a restricted option, it is ignored and a warning message is printed to the Cluster Operator log file.
 All other supported options are passed to Kafka.
 
-Use the three allowed `ssl` configuration options for client connection using a specific _cipher suite_ for a TLS version.
+There are exceptions to the restrictions imposed on the configuration.
+You can use three allowed `ssl` configuration options for client connection using a specific _cipher suite_ for a TLS version.
 A _cipher suite_ combines algorithms for secure connection and data transfer.
```
Comment (laidan6000) [inline documentation/modules/ref-kafka-broker-configuration.adoc:40]: No need for italics on second use.

File: documentation/modules/ref-kafka-broker-configuration.adoc [hunk 3]
```diff
@@ -35,8 +35,10 @@ Specifically, you are prevented from modifying all configuration options with ke
...
+You can also configure the `zookeeper.connection.timeout.ms` property to set maximum time allowed for establishing a ZooKeeper connection.
```
Comment (laidan6000) [inline documentation/modules/ref-kafka-broker-configuration.adoc:41]: _Suggested change:_

maximum time > the maximum time

File: documentation/modules/ref-kafka-connect-configuration.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/ref-kafka-connect-configuration.adoc:45]: Updated in the 4 locations

File: documentation/modules/ref-kafka-broker-configuration.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/ref-kafka-broker-configuration.adoc:40]: Updated in all places

**Pull request merged**